File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/ce/KeyUsage.java
package aladdin.asn1.iso.pkix.ce;

public final class KeyUsage 
{
	public static final long NONE					= 0x0000;  
	public static final long DIGITAL_SIGNATURE		= 0x0001;  
	public static final long NON_REPUDIATION		= 0x0002; 
	public static final long KEY_ENCIPHERMENT		= 0x0004;
	public static final long DATA_ENCIPHERMENT		= 0x0008;
	public static final long KEY_AGREEMENT			= 0x0010;
	public static final long CERTIFICATE_SIGNATURE	= 0x0020;
	public static final long CRL_SIGNATURE			= 0x0040;
	public static final long ENCIPHER_ONLY			= 0x0080;
	public static final long DECIPHER_ONLY			= 0x0100;  
}

File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/CE/KeyUsage.cs
using System; 

namespace Aladdin.ASN1.ISO.PKIX.CE
{
	[Flags]
	public enum KeyUsage : long
	{
		None						= 0x0000,  
		DigitalSignature			= 0x0001,  
		NonRepudiation				= 0x0002, 
		KeyEncipherment				= 0x0004,
		DataEncipherment			= 0x0008,
		KeyAgreement				= 0x0010,
		KeyExchange					= 0x0014,
		CertificateSignature		= 0x0020,
		CrlSignature				= 0x0040,
		DataSignature				= 0x0061,  
		EncipherOnly				= 0x0080,
		DecipherOnly				= 0x0100 
	}
}


